File: android-native/app/src/main/java/flexr/social/app/ui/components/PhotoLightbox.kt
package flexr.social.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.pager.HorizontalPager
import androidx.compose.foundation.pager.rememberPagerState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.unit.dp
import androidx.compose.ui.window.Dialog
import androidx.compose.ui.window.DialogProperties
import coil.compose.AsyncImage
import flexr.social.app.core.designsystem.icon.FlexrIcons
import flexr.social.app.core.designsystem.theme.FlexrTheme
import flexr.social.app.core.designsystem.theme.MonoStyle
import flexr.social.app.domain.model.Photo

/**
 * Foto-Vollbild mit Wischgalerie.
 *
 * Nativ per [HorizontalPager] gelöst: Wischen, Fangpunkte und Vorausladen
 * kommen vom Framework — im Web war das handgeschriebene Touch-Logik.
 * Beim Schließen wird der zuletzt betrachtete Index zurückgemeldet, damit die
 * Karte dahinter dasselbe Foto zeigt.
 */
@Composable
fun PhotoLightbox(
    photos: List<Photo>,
    startIndex: Int,
    onClose: (lastIndex: Int) -> Unit,
) {
    if (photos.isEmpty()) return
    val pagerState = rememberPagerState(
        initialPage = startIndex.coerceIn(0, photos.lastIndex),
        pageCount = { photos.size },
    )

    Dialog(
        onDismissRequest = { onClose(pagerState.currentPage) },
        properties = DialogProperties(usePlatformDefaultWidth = false),
    ) {
        Box(
            Modifier
                .fillMaxSize()
                .background(Color.Black.copy(alpha = 0.96f)),
        ) {
            HorizontalPager(
                state = pagerState,
                modifier = Modifier.fillMaxSize(),
                pageSpacing = 12.dp,
            ) { page ->
                AsyncImage(
                    model = photos[page].url,
                    contentDescription = "Foto ${page + 1} von ${photos.size}",
                    contentScale = ContentScale.Fit,
                    modifier = Modifier
                        .fillMaxSize()
                        .padding(horizontal = 16.dp, vertical = 64.dp),
                )
            }

            IconButton(
                onClick = { onClose(pagerState.currentPage) },
                modifier = Modifier
                    .align(Alignment.TopEnd)
                    .padding(16.dp)
                    .size(42.dp)
                    .clip(CircleShape)
                    .background(Color.White.copy(alpha = 0.08f)),
            ) {
                Icon(FlexrIcons.Close, contentDescription = "Schließen", tint = Color.White)
            }

            if (photos.size > 1) {
                Row(
                    Modifier
                        .align(Alignment.BottomCenter)
                        .fillMaxWidth()
                        .padding(bottom = 26.dp),
                    horizontalArrangement = Arrangement.Center,
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text(
                        text = "${pagerState.currentPage + 1} / ${photos.size}",
                        style = MonoStyle,
                        color = FlexrTheme.colors.chalkDim,
                        modifier = Modifier
                            .clip(CircleShape)
                            .background(Color.White.copy(alpha = 0.08f))
                            .padding(horizontal = 12.dp, vertical = 6.dp),
                    )
                }
            }
        }
    }
}

File: ios/FLEXR/UI/Components/PhotoLightbox.swift
import SwiftUI

/// Foto-Vollbild mit Wischgalerie.
///
/// Nativ über eine seitenweise scrollende `TabView` gelöst: Wischen, Fangpunkte
/// und Seitenanzeige kommen vom Framework — im Web war das handgeschriebene
/// Touch-Logik. Beim Schließen wird der zuletzt betrachtete Index
/// zurückgemeldet, damit die Karte dahinter dasselbe Foto zeigt.
struct PhotoLightbox: View {

    let photos: [Photo]
    let startIndex: Int
    let onClose: (Int) -> Void

    @State private var index: Int

    init(photos: [Photo], startIndex: Int, onClose: @escaping (Int) -> Void) {
        self.photos = photos
        self.startIndex = startIndex
        self.onClose = onClose
        _index = State(initialValue: min(max(startIndex, 0), max(photos.count - 1, 0)))
    }

    var body: some View {
        ZStack {
            Color.black.opacity(0.96).ignoresSafeArea()

            TabView(selection: $index) {
                ForEach(photos.indices, id: \.self) { offset in
                    let photo = photos[offset]
                    PhotoImage(
                        source: .remote(photo.url),
                        contentMode: .fit,
                        accessibilityLabel: "Foto \(offset + 1) von \(photos.count)"
                    )
                    .padding(.horizontal, 16)
                    .padding(.vertical, 64)
                    .tag(offset)
                }
            }
            .tabViewStyle(.page(indexDisplayMode: .never))

            VStack {
                HStack {
                    Spacer()
                    Button { onClose(index) } label: {
                        Image(systemName: FlexrIcon.close)
                            .font(.system(size: 17, weight: .semibold))
                            .foregroundStyle(.white)
                            .frame(width: 42, height: 42)
                            .background(Circle().fill(.white.opacity(0.08)))
                    }
                    .buttonStyle(.plain)
                    .accessibilityLabel("Schließen")
                }
                .padding(16)

                Spacer()

                if photos.count > 1 {
                    Text("\(index + 1) / \(photos.count)")
                        .flexrText(.mono)
                        .foregroundStyle(FlexrColor.chalkDim)
                        .padding(.horizontal, 12)
                        .padding(.vertical, 6)
                        .background(Capsule().fill(.white.opacity(0.08)))
                        .padding(.bottom, 26)
                }
            }
        }
    }
}
